include/boost/url/impl/params.hpp

```diff
@@ -76,146 +89,33 @@ class params::iterator
         return tmp;
     }
 
-    BOOST_URL_DECL
     reference
-    operator*() const;
-
-    friend
-    bool
-    operator==(
-        iterator a,
-        iterator b) noexcept
-    {
-        BOOST_ASSERT(a.u_ == b.u_);
-        return a.u_ == b.u_ &&
-            a.i_ == b.i_;
-    }
-
-    friend
-    bool
-    operator!=(
-        iterator a,
-        iterator b) noexcept
-    {
-        BOOST_ASSERT(a.u_ == b.u_);
-        return a.u_ != b.u_ ||
-            a.i_ != b.i_;
-    }
-
-    // LegacyRandomAccessIterator
-
-    iterator&
-    operator+=(
-        ptrdiff_t n) noexcept
-    {
-        i_ += n;
-        return *this;
-    }
-
-    friend
-    iterator
-    operator+(
-        iterator it,
-        ptrdiff_t n) noexcept
-    {
-        return {
-            *it.u_,
-            it.i_ + n,
-            it.a_ };
-    }
-
-    friend
-    iterator
-    operator+(
-        ptrdiff_t n,
-        iterator it) noexcept
-    {
-        return {
-            *it.u_,
-            it.i_ + n,
-            it.a_ };
-    }
-
-    iterator&
-    operator-=(
-        ptrdiff_t n) noexcept
-    {
-        i_ -= n;
-        return *this;
-    }
-
-    friend
-    iterator
-    operator-(
-        iterator it,
-        ptrdiff_t n) noexcept
+    operator*() const
     {
+        params_view::reference r0 =
+            impl_.dereference();
         return {
```

Comment:
`params_view::iterator` and `params::iterator` use different types. We can either make this conversion or make `params::reference` be an alias to `params_view::reference`.

Reply:
If they use different types then they can't share the `dereference` code can they

Reply:
The logic is 99% the same. They are different types with exactly the same members. It's just a matter of converting `params_view::reference` to `params::reference`.

We can

- make this conversion where it is,
- make this conversion in another one-line dereference function, or
- make params::reference just be an alias to params_view::reference, since they have the exact members